cmd/skopeo/sync.go

```diff
@@ -277,16 +359,26 @@ func syncHandler(c *cli.Context) (retErr error) {
 	//TODO: should we assume that's our default manifest type?
 	manifestType := manifest.DockerV2Schema2MediaType
 
-	toCopy := imagesBySource{}
+	if c.IsSet("src-cert-dir") || c.IsSet("src-tls-verify") || c.IsSet("src-creds") {
+		return fmt.Errorf("Must either specify a '--source-file' or '--src-' flags")
+	}
+
+	var toCopyList []imagesBySource
 
 	if c.IsSet("source") {
-		toCopy, err = syncSourceHandler(c, destRef)
+		toCopy, err := syncSourceHandler(c, destRef)
+		if err != nil {
+			return err
+		}
+		toCopyList = append(toCopyList, toCopy)
```

Comment:
This is a bit redundant, the whole `{ ... }` block can be rewritten this way:

```go
toCopyList, retErr = syncSourceHandler(c, destRef)
if retErr != nil {
   return
}
```

Reply:
so you are proposing to also have syncSourceHandler return a slice?